src/stt/src/pdr/src/pd.cpp

```diff
@@ -72,17 +75,23 @@ static vector<Neighbors> get_nearest_neighbors(const vector<Point>& pts)
   // this coordinate would have another node in its bbox and is
   // therefore not a nearest neighbor.  This depends on processing the
   // nodes in order of increasing y distance.
-  vector<int> ur(pt_count, std::numeric_limits<int>::max());
-  vector<int> ul(pt_count, std::numeric_limits<int>::min());
-  vector<int> lr(pt_count, std::numeric_limits<int>::max());
-  vector<int> ll(pt_count, std::numeric_limits<int>::min());
+  data.reserve(pt_count * 5);
+  data.resize(pt_count, std::numeric_limits<int>::max());
+  data.resize(pt_count * 2, std::numeric_limits<int>::min());
+  data.resize(pt_count * 3, std::numeric_limits<int>::max());
+  data.resize(pt_count * 4, std::numeric_limits<int>::min());
+  data.resize(pt_count * 5);
```

Comment:
Why resize repeatedly - why not resize once (no need to reserve) and then use std::fill.

Reply:
`resize` does zero initialization. It's not much of a difference (maybe it even gets optimized away?) so I can change it if you insist.

Reply:
That's a fair point.  You could reorder the blocks so that you only resize once for the max() values and once for the min() values.

Reply:
Good idea, done.